R/apply_delay.R

```diff
@@ -74,40 +74,37 @@ apply_delay <- function(rep_tri_to_nowcast, delay_pmf) {
 
 #' Calculate the updated rows of the expected nowcasted triangle
 #'
-#' @param delay_index Integer indicating the delay index
+#' @param index Integer indicating the delay index
 #' @param expectation Matrix of the incomplete reporting matrix
-#' @param delay_prob Probability of a case being reported with the current delay
-#' @param delay_cdf_prev CDF of the delay PMF up to the previous delay
+#' @param delay_pmf Vector of delays assumed to be indexed starting at the
+#'   first delay column in `rep_mat_to_nowcast`
+#' @param delay_cdf Vector of cumulative probabilities of the delay PMF
 #' @param n_rows Number of rows in the expectation matrix
 #' @returns Matrix with another set of entries corresponding to the updated
 #'   values for the specified rows and column
 #' @keywords internal
 .calc_expectation <- function(
-    delay_index,
+    index,
     expectation,
-    delay_prob,
-    delay_cdf_prev,
+    delay_pmf,
```

Comment:
I think this sort of "copied" what I was experimenting with in #146, and I don't thin the changes it made to ingest the entire cdf and pmf are necessary... 

Since it always will return the expectation updated (or not updated) for a single index, we can continue to pass in a scalar value, `delay_prob` and `delay_cdf_prev` rather than doing that within the `.calc_expectation` function 

It doesn't matter a ton, but it maintains more of the previous functionality and spirit of the loop.

Reply:
yup good point. This is from an interim fix where it modified the while loop but keep it . Droppigh